pkg/workflow/engineimpl.go

```diff
@@ -247,21 +251,25 @@ func (e *EngineImpl) InvokeWithInputAndConfig(
 
 			// prepare logger
 			prefix := fmt.Sprintf("%s:%d", id.Host, e.invocationCounter)
-			e.mu.Unlock()
-
 			zlogger := e.logger.With().Str("ext", prefix).Logger()
 
+			localConfig := e.config
+			localNetwork := e.networkAccess
+			localUi := e.ui
+			localAnalytics := e.analytics
+
 			// prepare configuration
 			if config == nil {
-				config = e.config.Clone()
+				config = localConfig.Clone()
 			}
 
 			// prepare networkAccess
-			networkAccess := e.networkAccess.Clone()
+			networkAccess := localNetwork.Clone()
 			networkAccess.SetConfiguration(config)
+			e.mu.Unlock()
 
 			// create a context object for the invocation
-			context := NewInvocationContext(id, config, e, networkAccess, zlogger, e.analytics, e.ui)
+			context := NewInvocationContext(id, config, e, networkAccess, zlogger, localAnalytics, localUi)
```

Comment:
why a new network access, if you already have the `localNetwork`? Confused me briefly, why `networkAccess` is passed instead of `localNetwork`